### ISSUE TITLE
java.util.NoSuchElementException: HttpStreamsClientHandler#0-body-publisher

### ISSUE THREAD

**Issue by yzarubin** [aws/aws-sdk-java-v2#461]
I know there was a pull request for this issue in the past: https://github.com/aws/aws-sdk-java-v2/pull/411 I checked and the version of the sdkv2 I'm using does indeed have the fix.

I got this exception quite a few times. It happens only under very high load. I was able to generate these errors by using SDKv2 with OpenSSL (though I think they may occur without it?), and running a stress test on m5.xlarge, where I more or less saturated the NIC by PUTing lots of large objects (50 MiB+) to S3. This error is painful in particular, because it looks like it gets swallowed, and the CompletableFuture for the PUT never completes.

```
[aws-java-sdk-NettyEventLoop-0-7] ERROR software.amazon.awssdk.http.nio.netty.internal.ResponseHandler - Exception processing request: software.amazon.awssdk.http.DefaultSdkHttpFullRequest@79762b71
java.util.NoSuchElementException: HttpStreamsClientHandler#0-body-publisher
	at io.netty.channel.DefaultChannelPipeline.getContextOrDie(DefaultChannelPipeline.java:1089)
	at io.netty.channel.DefaultChannelPipeline.remove(DefaultChannelPipeline.java:443)
	at com.typesafe.netty.http.HttpStreamsHandler.removeHandlerIfActive(HttpStreamsHandler.java:328)
	at com.typesafe.netty.http.HttpStreamsHandler.handleReadHttpContent(HttpStreamsHandler.java:189)
	at com.typesafe.netty.http.HttpStreamsHandler.channelRead(HttpStreamsHandler.java:165)
	at com.typesafe.netty.http.HttpStreamsClientHandler.channelRead(HttpStreamsClientHandler.java:148)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:438)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:284)
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:253)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1380)
	at io.netty.handler.ssl.SslHandler.decodeNonJdkCompatible(SslHandler.java:1171)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1196)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:489)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:428)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:265)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1359)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:935)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:141)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:645)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:580)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:497)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:459)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:886)
	at java.lang.Thread.run(Thread.java:748)
```
## Your Environment
* AWS Java SDK version used: 2.0.0-preview-9
* JDK version used: jdk1.8.0_161.jdk
* Operating System and version: Mac Sierra 10.12.6, but also reproduced the problem on the newest EC2 Linux AMI's

**Comment by yzarubin** [comment]
@dagnir I'll try to give it another try this week, and I should have some reproducible code.

**Comment by varunnvs92** [comment]
@yzarubin 
Any luck with getting the repo code?

**Comment by zoewangg** [comment]
Closing due to no response. Please try with the latest released version (`2.1.0` as of today) and feel free to reopen if the issue still occurs.
